AttendanceSystem.Domain.Services/AttendanceService.cs

```diff
@@ -45,10 +45,17 @@ public async Task DeleteSession(Guid sessionId)
         await _context.SaveChangesAsync();
     }
 
-    public async Task SetAttendance(Guid sessionId, string studentId, AttendanceKind kind)
+    public async Task SetStudentAttendance(Guid sessionId, string studentId, AttendanceKind kind)
     {
         var session = await GetSession(sessionId);
-        session.SetAttendance(studentId, kind);
+        session.SetStudentAttendance(studentId, kind, DateTime.UtcNow);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task SetTeacherAttendance(Guid sessionId, string teacherId, AttendanceKind kind)
+    {
+        var session = await GetSession(sessionId);
```

Comment:
Does this work? I think it does not save correctly. In the GetSession, we don't have any include that includes the Register. We may want to create a private method that gets the session but includes the register for this method, just for optimalisation reasons.